torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -678,6 +670,58 @@ std::string FlightRecorder<EventType>::dump(
     result.insert(
         entries_key, getCollectiveTrace(includeStackTraces, onlyActive));
   }
+  return result;
+}
+
+template <typename EventType>
+std::string FlightRecorder<EventType>::dump_json(
+    bool includeCollectives,
+    bool onlyActive) {
+  return (get_dump_json(includeCollectives, onlyActive)).dump();
+}
+
+template <typename EventType>
+// dump all collectives + ncclDumpMap
+std::string FlightRecorder<EventType>::dump(
+    bool includeCollectives,
+    bool includeStackTraces,
+    bool onlyActive) {
+  return pickle_str(
+      get_dump(includeCollectives, includeStackTraces, onlyActive));
+}
+
+std::unique_ptr<DebugInfoWriter> DebugInfoWriter::writer_ = nullptr;
+std::atomic<bool> DebugInfoWriter::hasWriterRegistered_(false);
+
+float getDurationFromEvent(c10::Event& startEvent, c10::Event& endEvent) {
+  TORCH_CHECK(false, "getDuration not supported by c10::Event.");
+}
+
+#ifdef USE_C10D_NCCL
+template <typename EventType>
+std::string FlightRecorder<EventType>::dump_json(
+    const std::optional<std::unordered_map<
+        std::string,
+        std::unordered_map<std::string, std::string>>>& ncclDumpMap,
+    bool includeCollectives,
+    bool onlyActive) {
+  // Adding ncclDumpMap to the json object is NCCL specific.
+  auto result = get_dump_json(includeCollectives, onlyActive);
+  if (ncclDumpMap.has_value()) {
+    result[nccl_comm_key_str] = ncclDumpMap.value();
+  }
+  return result.dump();
+}
```

Comment:
nit: which class/variable requires `#ifdef USE_C10D_NCCL` ? `ncclDumpMap`? But it seems to be a generic type: `std::unordered_map<std::string, std::string>>>`

Reply:
currently ncclDumpMap is mostly used by ncclx, if we believe this is necessary, I won't remove it.